DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/stp/test_stp_select_root.py

```diff
@@ -0,0 +1,640 @@
+    ]}])
+    assert out[0][dent]['rc'] == 0, 'Failed to enable stp/rstp'
+
+    # 4. Set link up on all participant ports
+    out = await IpLink.set(input_data=[{dent: [
+        {'device': port,
+         'operstate': 'up'} for port in ports]}])
+    assert out[0][dent]['rc'] == 0, 'Failed to set port state UP'
+
+    out = await IpLink.set(input_data=[{dent: [
+        {'device': bridge,
+         'operstate': 'up'}]}])
+    assert out[0][dent]['rc'] == 0, 'Failed to set bridge to state UP'
+
+    dev_groups = tgen_utils_dev_groups_from_config(
+        {'ixp': port, 'ip': f'1.1.1.{idx}', 'gw': '1.1.1.5', 'plen': 24}
+        for idx, port in enumerate(tg_ports, start=1)
+    )
+    await tgen_utils_traffic_generator_connect(tgen_dev, tg_ports, ports, dev_groups)
+
+    # 5. Transmit BDPU with root-id=bridge-id=00:44:XX:XX:XX:XX" wait ~40 seconds until topology fully converges
+    stp = {
+        'rootIdentifier': f'8000{root_mac.replace(":", "")}',
+        'bridgeIdentifier': f'8000{root_mac.replace(":", "")}',
+        'portIdentifier': '8002',
+        'messageAge': 1 << 8,
+        'frameSize': 100,
+    }
+    rstp = {
+        'agreement': 1,
+        'forwarding': 1,
+        'learning': 1,
+        'portRole': PortRole.DESIGNATED.value,
+        'proposal': 1,
+        'protocol': '0027',
+    } if version == 'rstp' else {}
+    streams = {
+        # BPDU stream with higher prio than bridge from port#0
+        f'{ports[0]} {version}': {
+            'type': 'bpdu',
+            'version': version,
+            'ep_source': ports[0],
+            'rate': 1,  # pps
+            'allowSelfDestined': True,
+            'srcMac': root_mac,
+            **stp,
+            **rstp,
+        }
+    }
+    await tgen_utils_setup_streams(tgen_dev, None, streams)
+    await tgen_utils_start_traffic(tgen_dev)
+    await asyncio.sleep(40)
+    # don't stop
+
+    # 6. Verify the bridge is not a root bridge
+    out = await Mstpctl.show(input_data=[{dent: [
+        {'parameter': 'bridge',
+         'bridge': bridge,
+         'options': '-f json'}]}], parse_output=True)
+    assert out[0][dent]['rc'] == 0, 'Failed to get bridge detail'
+    assert out[0][dent]['parsed_output'][0]['root-port'] != '', f'Bridge { bridge} is not a root bridge'
+
+    # 7. Change the highest MAC bridge to a MAC with lower value then the root bridge.
+    rc, _ = await dent_device.run_cmd(f'ifconfig {bridge} hw ether {get_rand_mac("00:22:XX:XX:XX:XX")}')
+    assert rc == 0, 'Failed to change MAC address'
+
+    # 8. Verify the bridge is a root bridge
+    out = await Mstpctl.show(input_data=[{dent: [
+        {'parameter': 'bridge',
+         'bridge': bridge,
+         'options': '-f json'}]}], parse_output=True)
+    assert out[0][dent]['rc'] == 0, 'Failed to get bridge detail'
+    assert out[0][dent]['parsed_output'][0]['root-port'] == '', f'Bridge { bridge} is not a root bridge'
+
+
+@pytest.mark.parametrize('version', ['stp', 'rstp'])
```

Comment:
You are using this in every test in this file. You can move this to `pytestmark` at L22 and remove this from all tests.